### ISSUE TITLE
Timbre -> tools.logging appender

### ISSUE THREAD

**Issue by ptaoussanis** [ptaoussanis/timbre#104]
Timbre ships with a tools.logging->timbre adapter.

Going the other direction, https://github.com/palletops/log-config has a Timbre->tools.logging appender in https://github.com/palletops/log-config#logging-from-timbre-to-java-logging.

Might be handy to bring something similar to Timbre core? PRs welcome!

**Comment by fzakaria** [comment]
I've written [slf4j-timbre](https://github.com/fzakaria/slf4j-timbre) which directs all SLF4J to timbre.
If you use the bridges (log4j/jul/jcl/ etc..) to have them go through SLF4J  that means you can pipe all output to timbre!

**Comment by fzakaria** [comment]
It's more like `Everything -> Timbre`

**Comment by ptaoussanis** [comment]
This looks great Farid, thank you! (And sorry for the delay replying).

Would you perhaps like to make a PR to add your lib under https://github.com/ptaoussanis/timbre#3rd-party-tools-appenders-etc?

**Comment by ptaoussanis** [comment]
Or, sorry- was your intention to get this integrated directly into Timbre? I'd be happy with that too, whatever you prefer! Cheers :-)

**Comment by fzakaria** [comment]
Thanks. I've opened a PR. That might be the best bet otherwise I'd hate for you to maintain it however feel free to integrate directly if you want. I've found it invaluable (especially for web frameworks) that utilize Jetty (or any other java library that uses log4j/slf4j/jul)

**Comment by ptaoussanis** [comment]
This is a great addition, thanks! I'd love to see this in Timbre core at some point, just have my hands a bit full atm- so let's keep it as a linked lib for now and maybe look at fully integrating sometime in future if you're on board with that?

Cheers :-)

**Comment by fzakaria** [comment]
I'm on board.
I too am swamped at the moment (clojure is my off time fun!).

I'd love to figure out writing some tests first before porting it in, so
it's not a burden to maintain.
(Need to learn how to swap output stream in Timbre to a StringOutputStream)

If you have some code samples please send them my way when you have a
chance.
(My clojure-ness is still young)

FZ

Farid Zakaria

**Comment by ptaoussanis** [comment]
Hi Farid,

> (Need to learn how to swap output stream in Timbre to a StringOutputStream)

Sorry, could you maybe rephrase this? I don't understand.

So `(timbre/info "foo")` currently outputs to `*out*` (`System/out` by default). You'd like to redirect the output somewhere else? To where? Do you mean a ByteArrayOutputStream?

**Comment by fzakaria** [comment]
Yes, a ByteArrayOutputStream.

Farid Zakaria

**Comment by ptaoussanis** [comment]
Hi Farid, sorry for the delay responding - have been travelling.

Is this perhaps what you're looking for: `(with-out-str (timbre/info "foo"))`? If not, it may help if I had some context re: what specifically you're trying to accomplish? 

Thanks again, cheers! :-)

**Comment by ptaoussanis** [comment]
Think we can probably close this. Have just made the links to slf4j-timbre more prominent.

**Comment by arichiardi** [comment]
I was checking how to do exacly the `timbre` to `tools.logging` conversion and I actually am interested in porting that very good work in `log-config` over to timbre as standalone appender. I am asking on the the `log-config` repo if he does not mind. Should I reopen the issue?
